xapian-core/backends/remote/remote-database.cc

```diff
@@ -900,6 +900,52 @@ RemoteDatabase::remove_spelling(const string & word,
     return result;
 }
 
+TermList*
+RemoteDatabase::open_synonym_termlist(const string& word) const
+{
+    string message;
+    send_message(MSG_SYNONYMTERMLIST, word);
+    get_message(message, REPLY_SYNONYMTERMLIST);
+    return new RemoteMetadataTermList(string(), std::move(message));
+}
+
+TermList*
+RemoteDatabase::open_synonym_keylist(const string& prefix) const
+{
+    string message;
+    send_message(MSG_SYNONYMKEYLIST, prefix);
+    get_message(message, REPLY_SYNONYMKEYLIST);
+    return new RemoteMetadataTermList(string(), std::move(message));
```

Comment:
I think here you did want `prefix` not `string()` - the first argument is the common prefix on the entries, which means we can avoid the redundancy of sending that in the reply for every entry.

The difference compared to `RemoteDatabase::open_synonym_termlist()` is that there we're returning the synonyms for `word` (so there's no common prefix, at least not that the client knows in advance) whereas here we're returning all words with synonyms and which start with `prefix`.